R/read_fwf.R

```diff
@@ -88,27 +107,55 @@ fwf_empty <- function(file, skip = 0, col_names = NULL, comment = "", n = 100L)
 #' @param col_names Either NULL, or a character vector column names.
 fwf_widths <- function(widths, col_names = NULL) {
   pos <- cumsum(c(1, abs(widths)))
-
   fwf_positions(pos[-length(pos)], pos[-1] - 1, col_names)
 }
 
 #' @rdname read_fwf
 #' @export
 #' @param start,end Starting and ending (inclusive) positions of each field.
 #'    Use NA as last end field when reading a ragged fwf file.
-fwf_positions <- function(start, end, col_names = NULL) {
+fwf_positions <- function(start, end = NULL, col_names = NULL) {
 
   stopifnot(length(start) == length(end))
-
-  if (is.null(col_names)) {
-    col_names <- paste0("X", seq_along(start))
-  } else {
-    stopifnot(length(start) == length(col_names))
-  }
+  col_names <- fwf_col_names(col_names, length(start))
 
   list(
     begin = start - 1,
     end = end, # -1 to change to 0 offset, +1 to be exclusive,
     col_names = col_names
   )
 }
+
+
+#' @rdname read_fwf
+#' @export
+#' @param ... If the first element is a data frame,
```

Comment:
This feels too flexible to me. But if you really think it's a good idea to keep it, the function signature should be `x, ...`